master/buildbot/config.py

```diff
@@ -592,12 +613,20 @@ def load_www(self, filename, config_dict):
         allowed = set(['port', 'debug', 'json_cache_seconds',
                        'rest_minimum_version', 'allowed_origins', 'jsonp',
                        'plugins', 'auth', 'avatar_methods', 'logfileName',
-                       'logRotateLength', 'maxRotatedFiles'])
+                       'logRotateLength', 'maxRotatedFiles', 'versions'])
         unknown = set(www_cfg.iterkeys()) - allowed
         if unknown:
             error("unknown www configuration parameter(s) %s" %
                   (', '.join(unknown),))
 
+        if www_cfg.get('versions') is None:
+            www_cfg['versions'] = {}
+
+        if not isinstance(www_cfg['versions'], dict):
+            error("'versions' must be a dictionary")
+
+        www_cfg['env_versions'] = self.getEnvironmentVersions()
```

Comment:
I think that this bit, and the `getEnvironmentVerions` method, should be somewhere under `master/buildbot/www` instead of here in `master/buildbot/config.py`.